### ISSUE TITLE
[math] Big-endian buffer not supported on little-endian compiler

### ISSUE THREAD

**Issue by vallsv** [silx-kit/silx#1146]
Relative to https://github.com/silx-kit/silx/pull/1140

Numpy support operations on little and big endian arrays
```
In [9]: numpy.array([0, 1, 2, 3], dtype=">f") + numpy.array([0, 1, 2, 3], dtype="<f")
Out[9]: array([ 0.,  2.,  4.,  6.], dtype=float32)
````
Our math function should support different endianness
```
In [6]: silx.math.combo.min_max(numpy.array([0, 1, 2, 3], dtype="<f"))
Out[6]: <silx.math.combo._MinMaxResult at 0x57bf7b8>

In [7]: silx.math.combo.min_max(numpy.array([0, 1, 2, 3], dtype=">f"))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
E:\valls\silx\bootstrap.py in <module>()
----> 1 silx.math.combo.min_max(numpy.array([0, 1, 2, 3], dtype=">f"))

E:\valls\silx\silx\math\combo.pyx in silx.math.combo.min_max (silx\math\combo.c:14149)()
    316         return _finite_min_max(data, min_positive)
    317     else:
--> 318         return _min_max(data, min_positive)

E:\valls\silx\silx\math\combo.pyx in silx.math.combo._min_max (silx\math\combo.c:3652)()
    115 @cython.boundscheck(False)
    116 @cython.wraparound(False)
--> 117 def _min_max(_number[::1] data, bint min_positive=False):
    118     """:func:`min_max` implementation including infinite values
    119

ValueError: Big-endian buffer not supported on little-endian compiler
```

**Comment by t20100** [comment]
As we anyway make sure it is a contiguous array before going to C/C++/Cython, it should not be too much effort to make sure the data have a native endianness.

**Comment by t20100** [comment]
I reopen this issue.
All Cython modules are subject to this issue, so if we have to support endianness, let's do it everywhere...

**Comment by payno** [comment]
I had a look on it. 
Histogramnd and HistogramndLut had the same issue.
I haven't see other peace of code which can bring this issue. In most of case we are directly setting the type requested which will be native.
A second look would be better.

**Comment by vasole** [comment]
closed by #1220
